File: taskmanager/src/test/java/com/example/alexeyglushkov/taskmanager/task/StackTaskProviderTest.java
package com.example.alexeyglushkov.taskmanager.task;

import android.os.Handler;
import android.os.Looper;

import org.junit.Before;
import org.junit.Test;

/**
 * Created by alexeyglushkov on 13.08.16.
 */

public class StackTaskProviderTest {
    protected TaskPoolTestSet poolTest;
    protected TaskProviderTestSet providerTest;
    protected TaskProvider taskProvider;

    @Before
    public void setUp() throws Exception {
        taskProvider = prepareTaskProvider();

        poolTest = new TaskPoolTestSet();
        providerTest = new TaskProviderTestSet();

        preparePoolTest();
        prepareProviderTest();
    }

    protected TaskProvider prepareTaskProvider() {
        return new StackTaskProvider(false, new Handler(Looper.myLooper()), "TestId");
    }

    protected void preparePoolTest() {
        poolTest.before(taskProvider);
    }

    protected void prepareProviderTest() {
        providerTest.before(taskProvider);
    }

    // ProviderTests

    @Test
    public void testGetTopTaskWithBlockedTask() {
        providerTest.getTopTaskWithBlockedTask();
    }

    @Test
    public void testTakeTopTaskWithBlockedTask() {
        providerTest.takeTopTaskWithBlockedTask();
    }

    @Test
    public void testSetProviderId() {
        providerTest.setProviderId();
    }

    @Test
    public void testSetPriority() {
        providerTest.setPriority();
    }

    @Test
    public void testGetTopTaskWithoutFilter() {
        providerTest.getTopTaskWithoutFilter();
    }

    @Test
    public void testGetTopTaskWithFilter() {
        providerTest.getTopTaskWithFilter();
    }

    @Test
    public void testTakeTopTaskWithFilter() {
        providerTest.takeTopTaskWithFilter();
    }

    @Test
    public void testRemoveTaskWithUnknownType() {
        providerTest.removeTaskWithUnknownType();
    }

    // PoolTests

    @Test
    public void testSetGetHandler() {
        poolTest.setGetHandler();
    }

    @Test
    public void testAddTask() {
        poolTest.addTask();
    }

    @Test
    public void testAddTaskAddStatusListener() {
        poolTest.addTaskAddStatusListener();
    }

    @Test
    public void testAddStartedTask() {
        poolTest.addStartedTask();
    }

    @Test
    public void testRemoveTask() {
        poolTest.removeTask();
    }

    @Test
    public void testRemoveUnknownTask() {
        poolTest.removeUnknownTask();
    }

    @Test
    public void testGetTask() {
        poolTest.getTask();
    }

    @Test
    public void testGetUnknownTask() {
        poolTest.getUnknownTask();
    }

    @Test
    public void testGetTaskCount() {
        poolTest.getTaskCount();
    }

    @Test
    public void testGetTaskCount2() {
        poolTest.getTaskCount2();
    }

    @Test
    public void testSetGetUserData() {
        poolTest.setGetUserData();
    }

    @Test
    public void testAddStateListener() {
        poolTest.addStateListener();
    }

    @Test
    public void testRemoveStateListener() {
        poolTest.removeStateListener();
    }

    @Test
    public void testChangeTaskStatus() {
        poolTest.changeTaskStatus();
    }

    @Test
    public void testCheckTaskRemovingAfterFinishing() {
        poolTest.checkTaskRemovingAfterFinishing();
    }
}

File: taskmanager/src/test/java/com/example/alexeyglushkov/taskmanager/task/StackTaskProviderTest.kt
package com.example.alexeyglushkov.taskmanager.task

import com.example.alexeyglushkov.taskmanager.providers.StackTaskProvider
import com.example.alexeyglushkov.taskmanager.providers.TaskProvider
import org.junit.Before
import org.junit.Test

/**
 * Created by alexeyglushkov on 13.08.16.
 */

open class StackTaskProviderTest {
    protected lateinit var poolTest: TaskPoolTestSet
    protected lateinit var providerTest: TaskProviderTestSet
    protected lateinit var taskProvider: TaskProvider

    @Before
    @Throws(Exception::class)
    fun setUp() {
        taskProvider = prepareTaskProvider()

        poolTest = TaskPoolTestSet()
        providerTest = TaskProviderTestSet()

        preparePoolTest()
        prepareProviderTest()
    }

    protected open fun prepareTaskProvider(): TaskProvider {
        return StackTaskProvider(false, "TestId", TestTasks.createTestScopeThreadRunner())
    }

    protected fun preparePoolTest() {
        poolTest.before(taskProvider)
    }

    protected fun prepareProviderTest() {
        providerTest.before(taskProvider)
    }

    // ProviderTests

    @Test
    fun getTopTaskWithDependantTask() {
        providerTest.getTopTaskWithDependantTask()
    }

    @Test
    fun getTopTaskWithBlockedTask() {
        providerTest.getTopTaskWithBlockedTask()
    }

    @Test
    fun takeTopTaskWithDependantTask() {
        providerTest.takeTopTaskWithDependantTask()
    }

    @Test
    fun takeTopTaskWithBlockedTask() {
        providerTest.takeTopTaskWithBlockedTask()
    }

    @Test
    fun testSetProviderId() {
        providerTest.setProviderId()
    }

    @Test
    fun testSetPriority() {
        providerTest.setPriority()
    }

    @Test
    fun testGetTopTaskWithoutFilter() {
        providerTest.getTopTaskWithoutFilter()
    }

    @Test
    fun testGetTopTaskWithFilter() {
        providerTest.getTopTaskWithFilter()
    }

    @Test
    fun testTakeTopTaskWithFilter() {
        providerTest.takeTopTaskWithFilter()
    }

    @Test
    fun testRemoveTaskWithUnknownType() {
        providerTest.removeTaskWithUnknownType()
    }

    // PoolTests

    @Test
    fun testSetGetHandler() {
        poolTest.setGetHandler()
    }

    @Test
    fun testAddTask() {
        poolTest.addTask()
    }

    @Test
    fun addTaskCallsAddStatusListener() {
        poolTest.addTaskCallsAddStatusListener()
    }

    @Test
    fun removeTaskCallsRemoveTaskStatusListener() {
        poolTest.removeTaskCallsRemoveTaskStatusListener()
    }

    @Test
    fun testAddStartedTask() {
        poolTest.addStartedTask()
    }

    @Test
    fun testRemoveTask() {
        poolTest.removeTask()
    }

    @Test
    fun testRemoveUnknownTask() {
        poolTest.removeUnknownTask()
    }

    @Test
    fun testGetTask() {
        poolTest.getTask()
    }

    @Test
    fun testGetUnknownTask() {
        poolTest.getUnknownTask()
    }

    @Test
    fun testGetTaskCount() {
        poolTest.getTaskCount()
    }

    @Test
    fun testGetTaskCount2() {
        poolTest.getTaskCount2()
    }

    @Test
    fun testSetGetUserData() {
        poolTest.setGetUserData()
    }

    @Test
    fun testAddStateListener() {
        poolTest.addStateListener()
    }

    @Test
    fun testRemoveStateListener() {
        poolTest.removeStateListener()
    }

    @Test
    fun testChangeTaskStatus() {
        poolTest.changeTaskStatus()
    }

    @Test
    fun testCheckTaskRemovingAfterFinishing() {
        poolTest.checkTaskRemovingAfterFinishing()
    }

    @Test
    fun testCancelTask() {
        poolTest.cancelTask()
    }
}
